File: app/src/main/java/ezike/tobenna/myweather/data/local/dao/WeatherDao.java
package ezike.tobenna.myweather.data.local.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import ezike.tobenna.myweather.data.local.entity.WeatherResponse;

/**
 * @author tobennaezike
 */

@Dao
public interface WeatherDao {

    @Query("select * from weather_response")
    LiveData<WeatherResponse> getWeatherResponse();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertWeatherResponse(WeatherResponse weather);
}

File: app/src/main/java/ezike/tobenna/myweather/data/local/db/WeatherDao.kt
package ezike.tobenna.myweather.data.local.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import ezike.tobenna.myweather.data.model.WeatherResponse
import kotlinx.coroutines.flow.Flow

@Dao
interface WeatherDao {
    @Query("select * from weather_response")
    fun fetchWeather(): Flow<WeatherResponse>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(weather: WeatherResponse?)
}
